### DISCUSSION TITLE
Ability to implement or workaround VK_EXT_image_2d_view_of_3d?

### DISCUSSION THREAD

**Discussion by jarrodnorwell** [KhronosGroup/MoltenVK#2112] [Q&A]
Hello all,

I'm currently porting Yuzu to iPad and iPhone making use of MoltenVK for the Vulkan renderer and it's working quite well however, with Super Mario Bros. Wonder it's running into [2D views on 3D images](https://github.com/KhronosGroup/MoltenVK/blob/main/MoltenVK/MoltenVK/GPUObjects/MVKImage.mm#L1980). Looking at older issues on this repository there seems to be a workaround by using VK_*_COLOR_ATTACHMENT_BIT but that results in missing text in-game seen [here (5-20s)](https://youtu.be/-gVyVmhrQLw?si=5BlqgpCy_qUJbfw3&t=5).

Looking at the code for MVK there's a [comment](https://github.com/KhronosGroup/MoltenVK/blob/main/MoltenVK/MoltenVK/GPUObjects/MVKImage.mm#L1976) mentioning another potential workaround(?), is that the case and if yes, how would one go about implementing that fix?

**Comment by cdavis5e** [comment]
Is it *rendering* to a 3D image using a 2D image view, or is it *using* it from a shader? If it's the latter, nothing can be done, because that isn't supported under Metal. But if it's only rendering, even though it sets the usage to be more expansive (or doesn't set it at all on the view), we can do something about that.

> Looking at the code for MVK there's a [comment](https://github.com/KhronosGroup/MoltenVK/blob/main/MoltenVK/MoltenVK/GPUObjects/MVKImage.mm#L1976) mentioning another potential workaround(?)

That workaround is already implemented. See [here](https://github.com/KhronosGroup/MoltenVK/blob/ad8b963e7520aca14c70273e8ae89144f16699b3/MoltenVK/MoltenVK/GPUObjects/MVKImage.mm#L1906).
